tests/query/allocation_policies/test_allocation_policy_base.py

```diff
@@ -385,3 +386,35 @@ def test_bad_defaults() -> None:
         SomeParametrizedConfigPolicy(
             StorageKey("some_storage"), [], {"my_param_config": False}
         )
+
+
+@pytest.mark.redis_db
+@mock.patch("snuba.query.allocation_policies.AllocationPolicy")
```

Comment:
You're mocking the base class (which is under test) in the test for the base class. It's hard for me to understand how this test works or why it works. I think it would be much easier to do the following:

1. Make a policy for this test that errors on `_get_quota_allowance` and `_update_quota_balance` 
2. Make the policy inactive
3. see that you can run the inactive policy and it won't raise

For an example see `test_passes_through_on_error`

Reply:
By mocking the base class, we can assert if a method of the class (`_get_quota_allowance` and `_update_quota_balance`) is called inside the method we are testing. The goal is to ensure that these private methods (which are overridden by the child policies) are not called when the policy is inactive which this test checks. Does that make sense?

Reply:
but if you're mocking the whole base class, why does any of the functionality of the base class work?

Reply:
Ah you're right, I'm mocking the base class not the sub-class and since we've moved the test outside the sub-class (to the base) it doesn't make sense anymore. I've changed the test to what you've have suggested.